lib/std/debug.zig

```diff
@@ -110,25 +110,31 @@ pub fn getSelfDebugInfo() !*DebugInfo {
 }
 
 pub fn detectTTYConfig(file: std.fs.File) TTY.Config {
-    if (process.hasEnvVarConstant("ZIG_DEBUG_COLOR")) {
-        return .escape_codes;
-    } else if (process.hasEnvVarConstant("NO_COLOR")) {
+    if (builtin.os.tag == .wasi) {
+        // Per https://github.com/WebAssembly/WASI/issues/162 ANSI codes
+        // aren't currently supported.
         return .no_color;
     } else {
-        if (file.supportsAnsiEscapeCodes()) {
+        if (process.hasEnvVarConstant("ZIG_DEBUG_COLOR")) {
             return .escape_codes;
-        } else if (native_os == .windows and file.isTty()) {
-            var info: windows.CONSOLE_SCREEN_BUFFER_INFO = undefined;
-            if (windows.kernel32.GetConsoleScreenBufferInfo(file.handle, &info) != windows.TRUE) {
-                // TODO: Should this return an error instead?
+        } else if (process.hasEnvVarConstant("NO_COLOR")) {
+            return .no_color;
+        } else {
+            if (file.supportsAnsiEscapeCodes()) {
```

Comment:
Could you also replace the two `else { if`s in this function with `else if` when you handle the uses of `hasEnvVarConstant` in `main.zig`